templates/aws-stack.yml

```diff
@@ -921,6 +932,7 @@ Resources:
       Environment:
         Variables:
           BUILDKITE_AGENT_TOKEN: !Ref BuildkiteAgentToken
+          BUILDKITE_AGENT_TOKEN_SSM_KEY: !Ref BuildkiteAgentTokenPath
```

Comment:
Reading the source code of the lambda, this should be supported but it's a bit messy and still means the token is viewable in the lambda console so I think it would be better to set either env var dynamically. Let me know what you think

Reply:
I'd use an `!If UseSSMAgentToken` to blank out the value of `BUILDKITE_AGENT_TOKEN`.

Reply:
Is this the kind of thing you mean? Or literally an empty string?